React/nested-event-handling.md

```diff
@@ -180,187 +197,103 @@ The CSS is the most straightforward thing on our todo list, so let's get that ou
 }
 ```
 
-#### The Button
-Next, let's modify the `Student` component to include the `isPresent` property, and a button.
-
-```js
-// Student.js
-const Student = (props) => {
-
-return (
-  <section className='student'>
-    <h3>Student Component</h3>
-    <h4 className={(props.isPresent) ? 'present' : 'absent'}>Name {props.fullName} </h4>
-    <p>Email: {props.email} </p>
+#### Adding className for styling
 
+Next, let's modify the `Student` component to use the `present` property, to style the component.
 
-    <button
-    disabled={ this.props.isPresent }
-    >Mark Present</button>
-  </section>
-  );
-}
+```javascript
+// src/components/Student.js
+ // ...
+    <div className="student">
+      <h3 className={props.present ? 'present' : 'absent'}>{props.fullName}</h3>
+ // ...
 ```
 
-Change the initial state in the `StudentCollection` to verify this works as intended.
+Change the initial state in the `App` to verify this works as intended.
 
 #### The Event Handler
+
 Next, let's explore the event handler setup. What data do we need from the button click event to appropriately change the overall student's state?
 
-1. An Event Handler in the `Student.js` file
-1. A function that has access to the data (`this.state`) inside of `StudentCollection.js`
+1. An Event Handler in the `Student.js` file, we can use `onButtonClick` and `onFullNameInputChange`.
+1. A function that has access to the state inside of `App.js`.
 
-First we'll create the event handler function with an initial logging statement. We'll follow the same pattern as before for a DOM event handler: start with `on`, then indicate what element and what event are being handled.
+First we will modify the `onButtonClick` as follows.
 
 ```javascript
-// Student.js
-const Student = (props) => {
-  const onPresentButtonClick = () => {
-    console.log(props);
+const onButtonClick = () => {
+    const updatedStudent = {
+      fullName: props.fullName,
+      birthday: props.birthday,
+      email: props.email,
+      // Toggle present to a new value
+      present: !props.present,
+      id: props.id,
+    }
+
+    // call the function passed from `App`
+    props.onUpdateStudent(updatedStudent);
```

Comment:
This callback right here feels like a monumental new concept to students. Can we give more explanation to explicitly state this is a call to the function we created in `App`?

Reply:
There's quite a bit on this below, but I've added more.